src/big/big_float.cr

```diff
@@ -51,16 +83,26 @@ struct BigFloat < Float
     LibGMP.mpf_cmp(self, other)
   end
 
-  def <=>(other : Float)
+  def <=>(other : BigInt)
+    LibGMP.mpf_cmp_z(self, other)
+  end
+
+  def <=>(other : Float32 | Float64)
     LibGMP.mpf_cmp_d(self, other.to_f64)
   end
 
-  def <=>(other : Int::Signed)
-    LibGMP.mpf_cmp_si(self, other.to_i64)
+  def <=>(other : Int)
```

Comment:
I still do not believe Crystal overloading due to trouble found by @funny-falcon  :-)

So trying to not use it when possible with primitive supertypes and subtypes.